File: DSA/Recursion1b/RemoveX.java
import java.util.Scanner;

public class RemoveX {
	public static void main(String[] args) {
		Scanner s = new Scanner(System.in);
		String input = s.nextLine();
		System.out.println(solution.removeX(input));
	}
}

public class solution {

	// Return the changed string
    static int idx = 0;
	public static String removeX(String input){
		// Write your code here
       
        if(idx == input.length()){
            idx--;
            return "";
        }
        idx++;
        String str = removeX(input);
        if(idx>=0 && input.charAt(idx) != 'x'){
            str = input.charAt(idx) + str;
        }
        idx--;
        return str;
	}
}

File: BackToBasics/Recursion/remove_x.cpp
#include <iostream>
#include <string>
using namespace std;
void removeX(char *str)
{
    if (str[0] == '\0')
    {
        return;
    }
    if (str[0] == 'x')
    {
        int i = 1;
        for (; str[i] != '\0'; i++)
        {
            str[i - 1] = str[i];
        }
        str[i - 1] = '\0';
        removeX(str); // length is reduced above
    }
    else
    {
        removeX(str + 1);
    }
}

int main()
{
    // Write your code here
    char str[] = "xxabxzxc";
    removeX(str);
    cout << str << endl;
    return 0;
}